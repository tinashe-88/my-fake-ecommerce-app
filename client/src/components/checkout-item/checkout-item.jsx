/* eslint-disable jsx-a11y/accessible-emoji */
import React from 'react'
import { connect } from 'react-redux'

import { 
  addItem, 
  removeItem,
  clearItemFromCart
} from '../../redux/cart/cart.actions'

import {
  CheckoutItemContainer,
  ImageContainer,
  TextContainer,
  QuantityContainer,
  RemoveButtonContainer,
  MinusSignContainer,
  PlusSignContainer
} from './checkout-item.styles'

const CheckoutItem = ({ cartItem, clearItem, addItem, removeItem }) => {
  const { name, imageUrl, price, quantity } = cartItem
  return (
    <CheckoutItemContainer>
      <ImageContainer>
        <img src={imageUrl} alt="item"/>
      </ImageContainer>
      <TextContainer>{name}</TextContainer>
      <QuantityContainer>
        <MinusSignContainer onClick={() => removeItem(cartItem)}>
          &#10134;
        </MinusSignContainer>
        <span>{quantity}</span>
        <PlusSignContainer onClick={() => addItem(cartItem)}>
          &#10133;
        </PlusSignContainer>
      </QuantityContainer>
      <TextContainer>{price}</TextContainer>
      <RemoveButtonContainer onClick={() => clearItem(cartItem)}>
        &#10060;
      </RemoveButtonContainer>
    </CheckoutItemContainer>
  )
}

const mapDispatchToProps = dispatch => ({
  addItem: item => dispatch(addItem(item)),
  clearItem: item => dispatch(clearItemFromCart(item)),
  removeItem: item => dispatch(removeItem(item))
})

export default connect(null, mapDispatchToProps)(CheckoutItem)