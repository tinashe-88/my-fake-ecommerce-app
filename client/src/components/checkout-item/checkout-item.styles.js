import styled from 'styled-components'

export const CheckoutItemContainer = styled.div`
  width: 100%;
  display: flex;
  min-height: 100px;
  border-bottom: 1px solid darkgrey;
  padding: 15px 0;
  font-size: 20px;
  align-items: center;
  
  @media screen and (max-width: 800px) {
    font-size: 18px;
  }
`

export const TextContainer = styled.span`
  width: 23%;

  @media screen and (max-width: 800px) {
    width: 22%;
  }
`

export const ImageContainer = styled.div`
  width: 23%;
  padding-right: 15px;

  img {
    width: 100%;
    height: 100%;
  }
`

export const QuantityContainer = styled(TextContainer)`
  display: flex;

  span {
    margin: 0 10px;
  }

  div {
    cursor: pointer;
  }
`

export const RemoveButtonContainer = styled.span`
  padding-left: 33px;
  cursor: pointer;
`

RemoveButtonContainer.displayName = 'RemoveButtonContainer'

export const MinusSignContainer = styled.div`
  font-size: 15px;
`

MinusSignContainer.displayName = 'MinusSignContainer'

export const PlusSignContainer = styled.div`
  font-size: 15px;
`

PlusSignContainer.displayName = 'PlusSignContainer'