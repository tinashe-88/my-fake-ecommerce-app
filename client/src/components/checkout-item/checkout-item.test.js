import React from 'react'
import { shallow } from 'enzyme'

import { CheckoutItem } from './checkout-item'

describe('CheckoutItem component', () => {
  let wrapper
  let mockClearItem
  let mockAddItem
  let mockRemoveItem

  beforeEach(() => {
    mockClearItem = jest.fn()
    mockAddItem = jest.fn()
    mockRemoveItem = jest.fn()

    const mockProps = {
      cartItem: {
        imageUrl: 'www.images.com',
        price: 344,
        name: 'jackets',
        quantity: 4
      },
      clearItem: mockClearItem,
      addItem: mockAddItem,
      removeItem: mockRemoveItem,
    }

    wrapper = shallow(<CheckoutItem {...mockProps}/>)
  })

  it('Should render CheckoutItem component', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('Should call clearItem when RemoveButtonContainer button is clicked', () => {
    wrapper.find('RemoveButtonContainer').simulate('click')
    expect(mockClearItem).toHaveBeenCalled()
  })

  it('Should call removeItem when MinusSignContainer button is clicked', () => {
    wrapper.find('MinusSignContainer').simulate('click')
    expect(mockRemoveItem).toHaveBeenCalled()
  })

  it('Should call addItem when PlusSignContainer button is clicked', () => {
    wrapper.find('PlusSignContainer').simulate('click')
    expect(mockAddItem).toHaveBeenCalled()
  })
})
