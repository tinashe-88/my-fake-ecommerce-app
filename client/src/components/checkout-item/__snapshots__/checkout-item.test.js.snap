// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutItem component Should render CheckoutItem component 1`] = `
<styled.div>
  <styled.div>
    <img
      alt="item"
      src="www.images.com"
    />
  </styled.div>
  <styled.span>
    jackets
  </styled.span>
  <Styled(styled.span)>
    <MinusSignContainer
      onClick={[Function]}
    >
      ➖
    </MinusSignContainer>
    <span>
      4
    </span>
    <PlusSignContainer
      onClick={[Function]}
    >
      ➕
    </PlusSignContainer>
  </Styled(styled.span)>
  <styled.span>
    344
  </styled.span>
  <RemoveButtonContainer
    onClick={[Function]}
  >
    ❌
  </RemoveButtonContainer>
</styled.div>
`;
